pkg/destroy/aws/shared.go

```diff
@@ -101,26 +101,32 @@ func (o *ClusterUninstaller) removeSharedTag(ctx context.Context, session *sessi
 				o.Logger.Debugf("No matches in %s for %s: shared, removing client", *tagClient.Config.Region, key)
 				continue
 			}
-			nextTagClients = append(nextTagClients, tagClient)
 
 			for i := 0; i < len(arns); i += 20 {
 				request.ResourceARNList = make([]*string, 0, 20)
 				for j := 0; i+j < len(arns) && j < 20; j++ {
 					request.ResourceARNList = append(request.ResourceARNList, aws.String(arns[i+j]))
 				}
-				result, err := tagClient.UntagResourcesWithContext(ctx, request)
-				if err != nil {
-					err = errors.Wrap(err, "untag shared resources")
-					o.Logger.Info(err)
-					continue
-				}
-				for _, arn := range request.ResourceARNList {
-					if info, failed := result.FailedResourcesMap[*arn]; failed {
-						o.Logger.WithField("arn", *arn).Infof("Failed to remove tag %s: shared; error=%s", key, *info.ErrorMessage)
+				var result *resourcegroupstaggingapi.UntagResourcesOutput
+				for retry := 0; retry < 3; retry++ {
+					result, err = tagClient.UntagResourcesWithContext(ctx, request)
+					if err != nil {
+						err = errors.Wrap(err, "untag shared resources")
+						o.Logger.Info(err)
 						continue
 					}
-					o.Logger.WithField("arn", *arn).Infof("Removed tag %s: shared", key)
-					removed[*arn] = exists
+				}
+
+				if err == nil {
```

Comment:
Can/should we consider adding the error to the return value if the previous statements were successful?

Reply:
Wasn't there before. Could add it.

Reply:
I think the code below takes care of searching for the IAM roles that still exist and handle errors.